### PR TITLE
Make grpcio optional for proto-* packages.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Also, ensure that gapic packages require it.

### CODE REVIEW

**Comment by lukesneeringer** [comment]
Tested and proved to install the right things (with @geigerj watching) using `proto-google-cloud-datastore-v1` and `gapic-google-cloud-datastore-v1`, generated locally.

**Comment by lukesneeringer** [comment]
I am going to wait until #1069 is merged, and then I am going to resolve the conflicts and get another once-over on this, after which I will merge it. :-)

**Comment by lukesneeringer** [comment]
I am in the process of merging master into this branch and dealing with conflicts, but several things have changed. I have committed a work in progress commit which I plan to follow up on from home this evening.

**Do not merge.** :-)

**Comment by lukesneeringer** [comment]
Verification steps:

  * [ ] Generate `proto-google-cloud-datastore-v1` and `gapic-google-cloud-datastore-v1` using the usual `./execute_pipeline` commands.
  * [ ] Make a clean virtualenv (used `which python3` for the Python version)
  * [ ] Install `proto-google-cloud-datastore-v1` (generated locally from this toolkit PR) using `python setup.py install`.
  * [ ] Establish (with `pip freeze`) that `grpcio` _was not_ installed.
  * [ ] Install `gapic-google-cloud-datastore-v1` (generated locally from this toolkit PR) using `python setup.py install`.
  * [ ] Watch multiple screens of C warnings fly by, eyes glaze over.
  * [ ] Establish (with `pip freeze`) that `grpcio` _was_ installed.
  * :tada:

File: src/main/resources/com/google/api/codegen/py/setup.py.snip [hunk 1]
```diff
@@ -12,10 +12,16 @@
...
   install_requires = [
-      'googleapis-common-protos[grpc]>={@metadata.commonProtosVersionBound.lower}, <{@metadata.commonProtosVersionBound.upper}',
       'google-gax>={@metadata.gaxVersionBound.lower}, <{@metadata.gaxVersionBound.upper}',
-      '{@metadata.protoPackageName}[grpc]>={@metadata.packageVersionBound.lower}, <{@metadata.packageVersionBound.upper}',
       'oauth2client>={@metadata.authVersionBound.lower}, <{@metadata.authVersionBound.upper}',
+      '{@metadata.protoPackageName}[grpc]>={@metadata.packageVersionBound.lower}, <{@metadata.packageVersionBound.upper}',
+      @join packageDep : metadata.protoPackageDependencies
+        @if or(packageDep.name == "googleapis-common-protos", packageDep.name.startsWith("proto-"))
+          '{@packageDep.name}[grpc]>={@packageDep.versionBound.lower}, <{@packageDep.versionBound.upper}',
+        @else
```
Comment (geigerj) [inline src/main/resources/com/google/api/codegen/py/setup.py.snip]: When is this `else` case needed? I intended `metadata.protoPackageDependencies` to refer explicitly to proto packages when I created that variable.

Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/py/setup.py.snip]: The test case was `python-package`, and I was concerned that there would be a case where someone did not use it how you intended. Essentially, I do not want to get an error by referencing a `[grpc]` on a package that does not have one.

**Comment by lukesneeringer** [comment]
I see that you went back to approval on your review, so merging. Happy to change that `@else` case as a followup if requested.

**Comment by geigerj** [comment]
I didn't mark approval again -- I just posted a comment. If there's not a particular case that you had in mind, I'd prefer to remove the else case. It seems more likely that we'd introduce a proto package that doesn't start with the string `proto` in the future, than that we'd introduce a non-proto package into the `protoPackageDependencies` list.

**Comment by lukesneeringer** [comment]
Bah, must be a GitHub bug with regard to re-review. Happy to remove the `@else`; will send another PR.

**Pull request merged**